lib/validate.js

```diff
@@ -0,0 +1,29 @@
+module.exports = validate
+
+var Promise = require('lie')
+
+function validate (state, doc) {
+  var self = this
+  return new Promise(function (resolve, reject) {
+    var result
+
+    try {
+      result = state.validate.call(self, doc)
```

Comment:
I don’t think you need need to bind validate to `self` here, we try to avoid using `this` altogether in our codebase